balancer.go

```diff
@@ -48,87 +45,20 @@ type (
 		Connection   Connection
 	}
 
-	// Strategy represents the methods a strategy need's to implement
-	Strategy interface {
-		Balance() (*Node, error)
-		RemoveNode(node *Node)
-		AddNode(node *Node)
-	}
-
-	// RoundRobinPolicy implements a Round Robin policy
-	RoundRobinPolicy struct {
-		nodes Nodes
-		next  uint32
-		lock  sync.RWMutex
-	}
-
 	// Balancer represents SQL Load Balancer and wraps sql.ExecerContext and sql.QueryerContext compatible methods
 	// All methods exposed are balanced following the Strategy provided
 	Balancer struct {
-		nodes  Nodes
-		policy Strategy
+		nodes    Nodes
+		strategy Strategy
 	}
 )
 
-func NewRoundRobinPolicy(connections Nodes) RoundRobinPolicy {
-	return RoundRobinPolicy{connections, 0, sync.RWMutex{}}
-}
-
-func (p *RoundRobinPolicy) Balance() (*Node, error) {
-	p.lock.Lock()
-	defer p.lock.Unlock()
-	if len(p.nodes) == 0 {
-		return nil, fmt.Errorf("empty set of nodes")
+func NewStrategy(nodes Nodes, strategyType StrategyType) (Strategy, error) {
```

Comment:
i really dont see a point to this factory.

To me, `Config` would not have a strategy type. 
Instead, config could hold a `StrategyFactory`, which would receive nodes and return a strategy.

This could be a way to allow the community to develop any strategy implementation and initialization.

Thoughts?

Reply:
I like your suggestion. Will do that in a separated Pull request, since it is not in the scope of this one.